src/views/CreateUserAccountView/CreateUserAccountPage.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react';
+
+import { Redirect, useHistory } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import {
+  Box,
+  CircularProgress,
+  Container,
+  Typography,
+} from '@material-ui/core';
+import { toast } from 'react-toastify';
+import { useQuery } from '../../old/lib/hooks/useQuery';
+import { langTokens } from '../../locales/localizationInit';
+import { BasicButton, BasicInput } from '../../components/Form';
+import { passwordValidationObj } from '../../old/lib/components/Users/validationRules';
+import { useStyles } from './CreateUserAccountPage.style';
+import { activateUser } from '../../old/lib/utilities/API/api';
+import { LoadingStatusEnum, LoadingStatusType } from '../../old/lib/types';
+
+interface ICreateUserAccount {
+  newPassword: string;
+  matchPassword: string;
+}
+
+const CreateUserAccountPage = () => {
+  const query = useQuery();
+  const history = useHistory();
+  const { t } = useTranslation();
+
+  const classes = useStyles();
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, handleSubmit, errors, watch, formState } = useForm<
+    ICreateUserAccount
+  >({
+    mode: 'onTouched',
+  });
+
+  const [loadingStatus, setLoadingStatus] = useState<LoadingStatusType>(
+    LoadingStatusEnum.idle,
+  );
+
+  const onSubmit = async (inputs: ICreateUserAccount) => {
+    const token = query.get('token');
+    if (!token) {
+      toast.error('No token provided');
+      return;
+    }
+    try {
+      setLoadingStatus(LoadingStatusEnum.pending);
+      await activateUser(token, inputs.newPassword, inputs.matchPassword);
+      toast.success('User was activated');
+      setLoadingStatus(LoadingStatusEnum.succeeded);
+    } catch (err) {
+      toast.error('Error activating user');
+      setLoadingStatus(LoadingStatusEnum.idle);
+    }
+  };
+
+  const loadingView = (
+    <Box className="mainLoading">
+      <CircularProgress />
+    </Box>
+  );
+
+  const createUserAccSuccess = (
```

Comment:
I suggest to make `createUserAccSuccess` and `createUserAccForm` to be separate components. So `CreateUserAccountForm` would have form and its state while `CreateUserAccountPage` would display either a form, or success message + button, or loading indicator

Reply:
Okay, I will remake it